cpp/src/IceGrid/LocatorI.cpp

```diff
@@ -1043,22 +946,28 @@ LocatorI::getDirectProxyException(const LocatorAdapterInfo& adapter, const Ice::
 
     if(activate)
     {
-        for(PendingRequests::iterator q = requests.begin(); q != requests.end(); ++q)
+        for(const auto& request : requests)
         {
-            (*q)->activating(adapter.id);
+            request->activating(adapter.id);
         }
 
-        int timeout = adapter.activationTimeout + adapter.deactivationTimeout;
-        AdapterPrx::uncheckedCast(adapter.proxy->ice_invocationTimeout(timeout * 1000))->begin_activate(
-            newCallback_Adapter_activate(new AdapterActivateCallback(this, adapter),
-                                         &AdapterActivateCallback::response,
-                                         &AdapterActivateCallback::exception));
+        int timeout = duration_cast<milliseconds>(adapter.activationTimeout + adapter.deactivationTimeout).count();
```

Comment:
VisualStudio 2019 complains about this

```
 88>D:\4.0\icegrid\cpp\src\IceGrid\LocatorI.cpp(954,21): error C2220: the following warning is treated as an error [D:\4.0\icegrid\cpp\src\IceGrid\msbuild\icegridnode\icegridnode.vcxproj]
    88>D:\4.0\icegrid\cpp\src\IceGrid\LocatorI.cpp(954,21): warning C4244: 'initializing': conversion from '_Rep' to 'int', possible loss of data [D:\4.0\icegrid\cpp\src\IceGrid\msbuild\icegridnode\
       icegridnode.vcxproj]
```

Reply:
I'll change it to `int timeout = secondsToInt(adapter.activationTimeout + adapter.deactivationTimeout) * 1000;`

Reply:
I think Windows uses and Int64 in 64 bit builds

Reply:
The `secondsToInt` helper takes care of converting this.